src/routes/organization/[organizationId]/+page.svelte

```diff
@@ -3,19 +3,43 @@
   import { page } from '$app/stores';
   import { createWritableStore, user } from '$lib/firebase';
   import type { PageData } from './$types';
-  import OrganizationMemberAdd from './OrganizationMemberAdd.svelte';
-  interface Members {
-    displayName: string;
-    email: string;
-    uid: string;
-  }
-
+  import InviteModal from './InviteModal.svelte';
+  import { invalidateAll } from '$app/navigation';
   export let data: PageData;
-  const members: Members[] = JSON.parse(data.members);
+  let members: Database.Organization.MemberDetails[];
+  let invites: Database.Invite.InviteDetails[];
+  $: members = JSON.parse(data.memberDetails);
+  $: invites = JSON.parse(data.inviteDetails);
   const { organizationId } = $page.params;
   const organization = createWritableStore<Database.Organization>(`/organizations/${organizationId}`);
 
   let showMemberAddModal = false;
+
+  const cancelInvite = async (inviteId: string) => {
+    await fetch(`${$page.url.href}/invites`, {
+      method: 'DELETE',
+      body: JSON.stringify([inviteId]),
+    });
+    invalidateAll();
+  };
+
+  const removeMember = async (uid: string) => {
+    await fetch(`${$page.url.href}/members`, {
+      method: 'DELETE',
+      body: JSON.stringify([uid]),
+    });
+    invalidateAll();
+  };
+
+  const promoteMember = (uid: string) => {
+    $organization!.private.members[uid].role = 'admin';
```

Comment:
isn't this cool!? Because of the way I setup `createWritableStore` in `$lib/firebase`, this will update the value in firebase. I want to use this more often

Reply:
honestly it really is cool, i definitely need to utilize the readable/writable stores more often, it's such a cool feature